e2e/orchestrator/version_test.go

```diff
@@ -0,0 +1,45 @@
+package orchestrator
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	shlex "github.com/flynn-archive/go-shlex"
+	"github.com/gotestyourself/gotestyourself/fs"
+	"github.com/gotestyourself/gotestyourself/icmd"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestVersionWithDefaultOrchestrator(t *testing.T) {
+	// Orchestrator by default
+	result := icmd.RunCmd(shell(t, "docker version"))
+	result.Assert(t, icmd.Success)
+	assert.Contains(t, result.Stdout(), "Orchestrator: swarm")
+}
+
+func TestVersionWithOverridenEnvOrchestrator(t *testing.T) {
+	// Override orchestrator using environment variable
+	result := icmd.RunCmd(shell(t, "docker version"), func(cmd *icmd.Cmd) {
+		cmd.Env = append(cmd.Env, append(os.Environ(), "DOCKER_ORCHESTRATOR=kubernetes")...)
+	})
+	result.Assert(t, icmd.Success)
+	assert.Contains(t, result.Stdout(), "Orchestrator: kubernetes")
+}
+
+func TestVersionWithOverridenConfigOrchestrator(t *testing.T) {
```

Comment:
https://github.com/docker/cli/blob/master/TESTING.md#guidelines

e2e tests are for commands. `orchestrator` is not a command. Please move one of these to `e2e/version`.

The rest should not be e2e tests. They are test client-only logic, so they should be unit tests.